### DISCUSSION TITLE
Allow custom subject lines for password request

### DISCUSSION THREAD

**Discussion by samuelfranzini** [directus/directus#23011] [Feature Requests]
### Summary

The `requestPasswordReset` function in the `users` service accepts `subject` as an optional field, only this parameter is not passed in the `/auth/password/request` API route parameters. 

I have no idea why, but I'd like to propose a pull request to implement this oversight.

### Basic Example

The basic API call is as follows:

```
await` api.post('auth/password/request', {
	email: "john@doe.tld",
	reset_url: "http://localhost:3000/reset-pwd",
});
```

The new one would be :

```
await api.post('auth/password/request', {
	email: "john@doe.tld",
	reset_url: "http://localhost:3000/reset-pwd",
        subject: "Hello, your requested new password !"
});
```

### Motivation

I need to be able to customize the title of the email that's sent out regarding the password reset request for non-English-speaking users in particular. 

### Detailed Design

The implementation has already been done on Directus, and only the function calling the service has been modified.

### Must Have:
- Custom emails subjects for request password

### Drawbacks

Subject can be treated as an optional field, allowing for backwards compatibility. The existing implementation of the request password feature can remain unchanged.

### Alternatives

N/A

### Adoption Strategy

Si d'autres utilisateurs souhaitent utiliser le point de terminaison d'invitation dans leurs extensions, ils peuvent désormais inclure un sujet. Cela correspondra à la façon dont le service de notification est déjà utilisé. L'adoption devrait être simple.

**Comment by samuelfranzini** [comment]
See my PR in https://github.com/directus/directus/pull/23012

**Comment by jekuer** [comment]
I built an extension to translate system emails.
It also allows to alter the subject of the default language.
-> https://github.com/jekuer/directus-extension-system-email-i18n
